### PR TITLE
etcdctl: added domain discovery flag

### LINKED ISSUE #2636: DNS discovery for etcdctl
It would be nice to have the possibility to use the same DNS discovery with etcdctl as with confd (which is the SRV entry _etcd._tcp.example.com). It could be something like `etcdctl -D example.com ...`. That's fault-tolerant (one must not know which hosts are up or even how they are called) and much faster typing than `etcdctl -C http://etcd1.example.com:2379,http://etcd2.example.com:2379 ...`.

### PR DESCRIPTION
**Pull request by MSamman**
provided a domain, will look up SRV records for etcd endpoints

Fixes #2636

### CODE REVIEW

File: etcdctl/README.md [hunk 1]
```diff
@@ -244,6 +244,16 @@ etcdctl --peers http://10.0.28.1:4002 my-key to-a-value
 etcdctl --peers http://10.0.28.1:4002,http://10.0.28.2:4002,http://10.0.28.3:4002 etcdctl set my-key to-a-value
 ```
 
+## DNS Discovery
+
+If you want to discover your etcd cluster through domain SRV records you can specify
+a `--discovery-domain` flag or `ETCDCTL_DISCOVERY_DOMAIN` environment variable.
```
Comment (xiang90) [inline etcdctl/README.md]: --discovery-srv ?

**Comment by xiang90** [comment]
The overall approach looks good to me. I think we should change `--discovery-domain` to `--disocvery-srv`. Also we need to document that the discovery's priority is higher than peer list.

BTW, have you tested it?

**Comment by MSamman** [comment]
I tested this by setting up a domain name with some SRV records and then trying to set/get keys using the domain name. However, I think I setup etcd incorrectly as I was getting a `404 page not found` response back from the server. Maybe because the server was running etcd 0.4.9 and not 2.0.0?  (I am an etcd noob) This was happening when I used the `--discovery-domain` option and also the `--peers` option. I was unable to set/get keys due to this issue.

**Comment by xiang90** [comment]
@MSamman You can enable `--debug` flag on etcdctl command. Then you can see the HTTP request etcdctl is sending. I think it would help you to debug the issue. Also I suggest you to try etcd 2.0 instead of 0.4.

**Comment by MSamman** [comment]
OK got it working and it looks good.

If you would like to test, I have etcd 2.0 running on a EC2 instance at `moesef.com`. I can get and set keys just find using `-D`, `--discovery-srv` or `ETCDCTL_DISCOVERY_SRV`. When the peers option is set along with these, it is ignored. Pull request updated per your original feedback.

**Comment by xiang90** [comment]
@MSamman LGTM

/cc @yichengq

**Comment by yichengq** [comment]
lgtm. Thanks for the contribution!

**Pull request merged**